smartmontools: 7.4 -> 7.5

Frostman opened the pull request:
Big update for the smartmontools. One of the biggest highlights is support for NVMe self-test and many new HDDs/SSDs added to the database.

Changelog: https://www.smartmontools.org/browser/tags/RELEASE_7_5/smartmontools/NEWS
Tickets: https://www.smartmontools.org/query?milestone=Release+7.5

I've tested it on a few of my systems in homelab and it looks good.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Frostman commented:
It would be great to get it into 25.05

bjornfor commented:
> Big update for the smartmontools. One of the biggest highlights is support for NVMe self-test and many new HDDs/SSDs added to the database.
>
> Changelog: https://www.smartmontools.org/browser/tags/RELEASE_7_5/smartmontools/NEWS

When reading git log, I'd like to see this info too. Why only write it for github? If written for git, the github PR message will get a copy of this info too.

bjornfor commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 403210`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 20 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_4_hardened.zfs_2_2</li>
    <li>linuxKernel.packages.linux_5_4_hardened.zfs_2_3</li>
    <li>linuxKernel.packages.linux_5_4_hardened.zfs_unstable</li>
    <li>linuxKernel.packages.linux_6_13.zfs_2_2</li>
    <li>linuxKernel.packages.linux_6_13_hardened.zfs_2_2</li>
    <li>linuxKernel.packages.linux_6_14.zfs_2_2</li>
    <li>linuxKernel.packages.linux_6_14.zfs_2_3</li>
    <li>linuxKernel.packages.linux_6_14.zfs_unstable</li>
    <li>linuxKernel.packages.linux_ham.zfs_2_2</li>
    <li>linuxKernel.packages.linux_latest_libre.zfs_2_2</li>
    <li>linuxKernel.packages.linux_latest_libre.zfs_2_3</li>
    <li>linuxKernel.packages.linux_latest_libre.zfs_unstable</li>
    <li>linuxKernel.packages.linux_lqx.zfs_2_2</li>
    <li>linuxKernel.packages.linux_lqx.zfs_2_3</li>
    <li>linuxKernel.packages.linux_lqx.zfs_unstable</li>
    <li>linuxKernel.packages.linux_xanmod_latest.zfs_2_2</li>
    <li>linuxKernel.packages.linux_xanmod_stable.zfs_2_2</li>
    <li>linuxKernel.packages.linux_zen.zfs_2_2</li>
    <li>linuxKernel.packages.linux_zen.zfs_2_3</li>
    <li>linuxKernel.packages.linux_zen.zfs_unstable</li>
  </ul>
</details>
<details>
  <summary>:x: 7 packages failed to build:</summary>
  <ul>
    <li>htcondor</li>
    <li>linuxKernel.packages.linux_5_10.zfs_2_2</li>
    <li>linuxKernel.packages.linux_5_10.zfs_2_3 (linuxKernel.packages.linux_5_10.zfs_unstable)</li>
    <li>python312Packages.py-libzfs</li>
    <li>python312Packages.py-libzfs.dist</li>
    <li>python313Packages.guestfs</li>
    <li>python313Packages.guestfs.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 134 packages built:</summary>
  <ul>
    <li>appvm</li>
    <li>booster</li>
    <li>ceph</li>
    <li>ceph-client</li>
    <li>ceph-csi</li>
    <li>ceph.dev</li>
    <li>ceph.doc</li>
    <li>libceph (ceph.lib ,libceph.dev ,libceph.doc ,libceph.lib ,libceph.man)</li>
    <li>ceph.man</li>
    <li>collectd</li>
    <li>diffoscope</li>
    <li>diffoscope.dist</li>
    <li>diffoscope.man</li>
    <li>docker-machine-kvm2</li>
    <li>easysnap</li>
    <li>facter</li>
    <li>fdroidserver</li>
    <li>fdroidserver.dist</li>
    <li>gnome-boxes</li>
    <li>gnomeExtensions.freon</li>
    <li>gsmartcontrol</li>
    <li>guestfs-tools</li>
    <li>hddfancontrol</li>
    <li>hw-probe</li>
    <li>kdePackages.kpmcore</li>
    <li>kdePackages.kpmcore.debug</li>
    <li>kdePackages.kpmcore.dev</li>
    <li>kdePackages.kpmcore.devtools</li>
    <li>kdePackages.partitionmanager</li>
    <li>kdePackages.partitionmanager.debug</li>
    <li>kdePackages.partitionmanager.dev</li>
    <li>kdePackages.partitionmanager.devtools</li>
    <li>kdePackages.plasma-disks</li>
    <li>kdePackages.plasma-disks.debug</li>
    <li>kdePackages.plasma-disks.dev</li>
    <li>kdePackages.plasma-disks.devtools</li>
    <li>libguestfs</li>
    <li>libguestfs-with-appliance</li>
    <li>libguestfs-with-appliance.guestfsd</li>
    <li>libguestfs.guestfsd</li>
    <li>librenms</li>
    <li>libsForQt5.kinfocenter</li>
    <li>libsForQt5.partitionmanager</li>
    <li>libvirt</li>
    <li>libvirt-glib</li>
    <li>libvirt-glib.dev</li>
    <li>libvirt-glib.devdoc</li>
    <li>libvmi</li>
    <li>libvmi.dev</li>
    <li>libvmi.lib</li>
    <li>linuxKernel.packages.linux_5_10_hardened.zfs_2_2</li>
    <li>linuxKernel.packages.linux_5_10_hardened.zfs_2_3 (linuxKernel.packages.linux_5_10_hardened.zfs_unstable)</li>
    <li>linuxKernel.packages.linux_5_15.zfs_2_2</li>
    <li>linuxKernel.packages.linux_5_15.zfs_2_3 (linuxKernel.packages.linux_5_15.zfs_unstable)</li>
    <li>linuxKernel.packages.linux_5_15_hardened.zfs_2_2</li>
    <li>linuxKernel.packages.linux_5_15_hardened.zfs_2_3 (linuxKernel.packages.linux_5_15_hardened.zfs_unstable)</li>
    <li>linuxKernel.packages.linux_5_4.zfs_2_2</li>
    <li>linuxKernel.packages.linux_5_4.zfs_2_3 (linuxKernel.packages.linux_5_4.zfs_unstable)</li>
    <li>linuxKernel.packages.linux_6_1.zfs_2_2</li>
    <li>linuxKernel.packages.linux_6_1.zfs_2_3 (linuxKernel.packages.linux_6_1.zfs_unstable)</li>
    <li>linuxKernel.packages.linux_6_12.zfs_2_2</li>
    <li>linuxKernel.packages.linux_6_12.zfs_2_3 (linuxKernel.packages.linux_6_12.zfs_unstable)</li>
    <li>linuxKernel.packages.linux_hardened.zfs_2_2 (linuxKernel.packages.linux_6_12_hardened.zfs_2_2)</li>
    <li>linuxKernel.packages.linux_hardened.zfs_2_3 (linuxKernel.packages.linux_6_12_hardened.zfs_2_3 ,linuxKernel.packages.linux_hardened.zfs_unstable)</li>
    <li>linuxKernel.packages.linux_6_13.zfs_2_3 (linuxKernel.packages.linux_6_13.zfs_unstable)</li>
    <li>linuxKernel.packages.linux_6_13_hardened.zfs_2_3 (linuxKernel.packages.linux_6_13_hardened.zfs_unstable)</li>
    <li>linuxKernel.packages.linux_6_1_hardened.zfs_2_2</li>
    <li>linuxKernel.packages.linux_6_1_hardened.zfs_2_3 (linuxKernel.packages.linux_6_1_hardened.zfs_unstable)</li>
    <li>linuxKernel.packages.linux_6_6.zfs_2_2</li>
    <li>linuxKernel.packages.linux_6_6.zfs_2_3 (linuxKernel.packages.linux_6_6.zfs_unstable)</li>
    <li>linuxKernel.packages.linux_6_6_hardened.zfs_2_2</li>
    <li>linuxKernel.packages.linux_6_6_hardened.zfs_2_3 (linuxKernel.packages.linux_6_6_hardened.zfs_unstable)</li>
    <li>linuxKernel.packages.linux_ham.zfs_2_3 (linuxKernel.packages.linux_ham.zfs_unstable)</li>
    <li>linuxKernel.packages.linux_libre.zfs_2_2</li>
    <li>linuxKernel.packages.linux_libre.zfs_2_3 (linuxKernel.packages.linux_libre.zfs_unstable)</li>
    <li>linuxKernel.packages.linux_xanmod.zfs_2_2</li>
    <li>linuxKernel.packages.linux_xanmod.zfs_2_3 (linuxKernel.packages.linux_xanmod.zfs_unstable)</li>
    <li>linuxKernel.packages.linux_xanmod_latest.zfs_2_3 (linuxKernel.packages.linux_xanmod_latest.zfs_unstable ,linuxKernel.packages.linux_xanmod_stable.zfs_2_3 ,linuxKernel.packages.linux_xanmod_stable.zfs_unstable)</li>
    <li>mgmt</li>
    <li>minikube</li>
    <li>multipass</li>
    <li>nagiosPlugins.check_smartmon</li>
    <li>nagiosPlugins.check_zfs</li>
    <li>nwipe</li>
    <li>ocamlPackages.ocaml_libvirt</li>
    <li>perl538Packages.SysVirt</li>
    <li>perl538Packages.SysVirt.devdoc</li>
    <li>perl540Packages.SysVirt</li>
    <li>perl540Packages.SysVirt.devdoc</li>
    <li>podman-bootc</li>
    <li>prometheus-libvirt-exporter</li>
    <li>prometheus-smartctl-exporter</li>
    <li>python312Packages.guestfs</li>
    <li>python312Packages.guestfs.dist</li>
    <li>python312Packages.libvirt</li>
    <li>python312Packages.libvirt.dist</li>
    <li>python312Packages.pysmart</li>
    <li>python312Packages.pysmart.dist</li>
    <li>python313Packages.libvirt</li>
    <li>python313Packages.libvirt.dist</li>
    <li>python313Packages.pysmart</li>
    <li>python313Packages.pysmart.dist</li>
    <li>qdiskinfo</li>
    <li>qemu_full</li>
    <li>qemu_full.debug</li>
    <li>qemu_full.doc</li>
    <li>qemu_full.ga</li>
    <li>rubyPackages.ruby-libvirt</li>
    <li>rubyPackages_3_1.ruby-libvirt</li>
    <li>rubyPackages_3_2.ruby-libvirt</li>
    <li>rubyPackages_3_4.ruby-libvirt</li>
    <li>samba4Full</li>
    <li>samba4Full.dev</li>
    <li>samba4Full.man</li>
    <li>sanoid</li>
    <li>scrutiny-collector</li>
    <li>smartmontools</li>
    <li>snapraid</li>
    <li>tlp</li>
    <li>vagrant</li>
    <li>virt-manager</li>
    <li>virt-manager-qt</li>
    <li>virt-top</li>
    <li>virt-v2v</li>
    <li>virt-viewer</li>
    <li>virtnbdbackup</li>
    <li>virtnbdbackup.dist</li>
    <li>zfs (zfs_unstable)</li>
    <li>zfs.dev (zfs_unstable.dev)</li>
    <li>zfs_2_2</li>
    <li>zfs_2_2.dev</li>
    <li>zfstools</li>
    <li>zpool-auto-expand-partitions</li>
    <li>zxfer</li>
  </ul>
</details>

bjornfor commented:
> ❌ 7 packages failed to build:
> 
>     * htcondor
> 
>     * linuxKernel.packages.linux_5_10.zfs_2_2
> 
>     * linuxKernel.packages.linux_5_10.zfs_2_3 (linuxKernel.packages.linux_5_10.zfs_unstable)
> 
>     * python312Packages.py-libzfs
> 
>     * python312Packages.py-libzfs.dist
> 
>     * python313Packages.guestfs
> 
>     * python313Packages.guestfs.dist

Already broken on the base branch.

bjornfor reviewed (approved):
Builds and runs.

bjornfor commented:
Needs rebase due to 509efdedbd1f2aceb3e719309cea3bcf99be7de0 ("treewide: move to by-name").

Frostman committed:
smartmontools: 7.4 -> 7.5

Big update for the smartmontools. One of the biggest highlights is support
for NVMe self-test and many new HDDs/SSDs added to the database.

Changelog: https://www.smartmontools.org/browser/tags/RELEASE_7_5/smartmontools/NEWS
Tickets: https://www.smartmontools.org/query?milestone=Release+7.5

Signed-off-by: Sergei Lukianov <me@slukjanov.name>
